components/ws-manager-bridge/src/bridge.ts

```diff
@@ -506,7 +512,12 @@ export class WorkspaceManagerBridge implements Disposable {
 
             for (const [instanceId, ri] of runningInstancesIdx.entries()) {
                 const instance = ri.latestInstance;
-                if (instance.status.phase !== "running") {
+                if (
+                    !(
+                        instance.status.phase === "running" ||
+                        durationLongerThanSeconds(Date.parse(instance.creationTime), maxTimeToRunningPhaseSeconds)
```

Comment:
If I understood correctly you are adding a timeout for the phase `pending`.
We handle timeouts for the various phases in [this method](https://github.com/gitpod-io/gitpod/blob/e184f5a68637e59169848bfefdb5ffbc81086f5e/components/ws-manager-bridge/src/bridge.ts#L573-L611). Would be interesting to understand why the case in [L583](https://github.com/gitpod-io/gitpod/blob/e184f5a68637e59169848bfefdb5ffbc81086f5e/components/ws-manager-bridge/src/bridge.ts#L583-L584) doesn't do it.

Reply:
@svenefftinge We have two controlling methods, which check instances in different phases, to clearly separate concerns. This was introduced after we detected outdated and broken logic in the old, single method. :upside_down_face: 

1. This one here is responsible for handling phases which [ws-manager](url) is responsible for, e.g. things that for whatever reason felt through the cracks, incl [running](url) and `pending`.
2. The one you linked is responsible for controlling phases WebApp is responsible for, e.g. `preparing` and `building`.

FYI: There is [a diagram](https://www.notion.so/gitpod/WebApp-Architecture-784cf14bc76440e3beb8de56f61e7340#6c9bd05eef5a44408171a95b14ae73a9) that tries to illustrate this a little bit.

Reply:
Ah, yes, that makes sense. I was confusing `pending` and `preparing`

Reply:
> I was confusing pending and preparing

Yeah, I agree. All of this feels more complex then it should :roll_eyes: 


@laushinka I'd really love to have at least a comment on this line which explains why we think this is an instance we do (not) want to stop.

Reply:
Looking at the logic now (btw. it's a little hard to wrap my head around the negated or, tbh 😇) it seems like it would also handle all `stopping` cases. Which is generally great as we have leaking stopping states, that we should handle.
But applying we should apply the timeout on top of stoppingTime instead of creationTime in this case.
Also I think this code would now also trigger for the phases webapp is responsible for if the the timeout is smaller than the respective timeout for the phase.

How about we explicitly list the phases we want to handle in this function (i.e. ws-manager responsibilities) and use a single timeout against the `max(CreationTime, StartedTime, StoppingTime)` (i.e. the last signal we got). The timeout doesn't really need to be configurable I think, but can be rather small and static. We mainly want to make sure we don't act on anything that just in this moment got removed from ws-manager (i.e. races).